Course work first

Alexsandr-SHay opened the pull request:
the first branch for editing the exam paper

Alexsandr-SHay committed:
empty commit

Alexsandr-SHay committed:
create new file Main

Alexsandr-SHay committed:
create new class ComicBook, ComicBookSalesman, PrintingData. create new enum GenreComics.
in Class ComicBookSalesman add Method addComic.

Добаленны классы Комикс, Продавец комиксов, класс реализующий ввод, проверку ввода и вывод данных PrintingData.
В класс комикс введены переменные класса, созданы конструкторы.
В Классе продавецКомиксов реализован метод добавления комикса с учётом ввода с клавиатуры.

Alexsandr-SHay committed:
Methods added in Class ComicBookSalesman:
deletingComicByName, deletingComicById, writeOffComicBookByName, writeOffComicByNameWithDataEntry, writeOffComicBookById, writeOffComicByIdWithDataEntry, editComicByNameDataEntry,editComicByIdDataEntry, searchObjectComicBook, searchObjectComicBook, searchForComicByTitle, searchForComicByTitleDataEntry, searchForComicByAuthor, searchForComicByAuthorDataEntry, searchForComicsByGenre, searchForComicsByGenreDataEntry.
Added a comic book database to check how the methods work.

Реализованы методы: Удаления комиксов(по имени и ID), Списывания комиксов(по имени и ID), редактирования комиксов(по имени и ID), поиска комикса по названию,автору, жанру(только методы вывода на экран).
Переделан метод добавления комикса.
Добавлена коллекция комиксов для тестирования методов.

Alexsandr-SHay committed:
The date of receipt parameter was added to the comic book class.
Methods added in Class ComicBookSalesman:
popularNewComicsForTheDay, popularNewComicsForTheWeek, popularNewComicsForTheMonth, popularNewComicsForTheYear, comicBookSales, comicBookSalesByName, comicBookSalesById, makeDiscountOnComic, makeDiscountOnComicByName, makeDiscountOnComicByID, makeDiscountOnALLComic, makeDiscountOnComicBookSeries, makeDiscountOnComicBookGenre, comicBookReservingById, listOfAllReservedComics
Class add Client.java

В класс комикса был добавлен параметр дата поступления.
Добавлен список проданных комиксов.
Реализованы методы: Поиска новинок по за различные периоды. Реализован метод продажи комикса по имени и ID с сохранением информации в отдельном списке. Реализованы методы объявления скидок по имени, ID, жанру, серии. Начата проработка над методом резервирования комиксов по конкретному клиенту(Для этого был создан отдельный класс Клиента.)

Удаления комиксов(по имени и ID), Списывания комиксов(по имени и ID), редактирования комиксов(по имени и ID), поиска комикса по названию,автору, жанру(только методы вывода на экран).
Переделан метод добавления комикса.
Добавлена коллекция комиксов для тестирования методов.

revers-back commented on .idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
еще папку out нужно и build

revers-back commented on .idea/ComicBookStore.iml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
это тоже в gitignore
для генерации удобно использовать сервер
https://www.toptal.com/developers/gitignore

revers-back commented on ComicBookStoreMyProject/src/Client.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.List;
+
+public class Client {
+    private static int idClient = 100_000;
```
не синхронизировано

revers-back commented on ComicBookStoreMyProject/src/ComicBook.java:
```diff
@@ -0,0 +1,170 @@
+    private int numberOfComics;                        // Количество комиксов в магазине / количество проданных комиксов
+    private LocalDate inputData;                       // Дата появления в магазине / дата продажи
+
+    public ComicBook(String comicBookName, String fullNameAuthor, String comicBookPublisher, int numberOfPages,
```
class Animal(val energy: Double, val weight: Double, val name: String)
множественные параметры форматируем каждый с новой строки
это уберегает от необходимости мерджить, если несколько человек в комнате одновременно добавят по одному или более параметров
data class Animal(
   val energy: Double,
   val weight: Double,
   val name: String
)

revers-back commented on ComicBookStoreMyProject/src/PrintingData.java:
```diff
@@ -0,0 +1,191 @@
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+public class PrintingData {                                     // Класс для печати и ввода информации
```
класс называться должен меню или PrintHelper 
сейчас "печатающий данные"

revers-back commented on ComicBookStoreMyProject/src/PrintingData.java:
```diff
@@ -0,0 +1,191 @@
+
+    private static Enum<GenreComics> checkGenreComics() {                // Выбор жанра комикса по вводу
+        String genreComic = scanner.nextLine();                          // todo пока до меня не доходит как можно
+        if (genreComic.equalsIgnoreCase("DETECTIVE") ||       // todo короче написать этот метод??
```
такое на кейс заменяют

revers-back commented on ComicBookStoreMyProject/src/PrintingData.java:
```diff
@@ -0,0 +1,191 @@
+        return yearOfPublication;
+    }
+
+    private static Enum<GenreComics> checkGenreComics() {                // Выбор жанра комикса по вводу
```
это можно внутри GenreComic положить добавии туда строку value

revers-back reviewed (changes requested):
в целом ок, стоит почитать про MVP паттерн, тема будет позже

revers-back commented:
каждый класс должен иметь комментарий в javadoc + каждый публичный метод

revers-back commented:
Прикрутить UI swing или JavaFx

Alexsandr-SHay committed:
Добавлены методы списания комиксов, продажи списаных комиксов, поисков коимксов по различным параметрам и времени.
Добавлены методы сохранения и загрузки списков. Позже будут перенесены в пакет repository
Внесены дополнительные методы в класс PrintHelper
